src/herder/TxSetFrame.cpp

```diff
@@ -33,23 +34,48 @@ namespace stellar
 
 using namespace std;
 
+void
+TxSetFrame::computeContentsHash()
+{
+    ZoneScoped;
+    sortInHashOrder();
+    SHA256 hasher;
+    hasher.add(mPreviousLedgerHash);
+    for (unsigned int n = 0; n < mTxsInHashOrder.size(); n++)
+    {
+        hasher.add(xdr::xdr_to_opaque(mTxsInHashOrder[n]->getEnvelope()));
+    }
+    mHash = hasher.finish();
+}
+
+TxSetFrame::TxSetFrame(Hash const& previousLedgerHash,
+                       Transactions const& transactions, bool inputIsApplyOrder)
+    : mPreviousLedgerHash(previousLedgerHash)
+    , mTxsInHashOrder(transactions)
+    , mTxsInApplyOrder(transactions)
+{
+    computeContentsHash();
+    if (!inputIsApplyOrder)
```

Comment:
I wonder how frequently is this actually true. I suspect it should be just a single code path. Maybe we should at least make this explicitly default argument in declaration?

Reply:
It is not used frequently, just a couple of places (simulation and some tests). 
However, I don't think having default arguments is a good practice (e.g. see this [blog](https://quuxplusone.github.io/blog/2020/04/18/default-function-arguments-are-the-devil/)) and I don't see them anywhere in our code base. I'm open to it though.

Reply:
I don't see anything wrong with default arguments in normal usage scenarios (the blog goes into pretty convoluted edge cases). I guess in this particular case my issue is rather not about using/not using the default arguments, but more about the fact that we kind of leak the simulation-related code (as we only need this to be `true` for simulation, right?) into the generic class. This causes the weird `false` argument to be present in multiple places.
Besides the virtual function approach I suggested in the comment above, another approach could be to create the `TxSetFrame` from hash-sorted and apply-sorted txs and have helper functions that build it for regular and simulation cases (similar to what we do for `TransactionFrame` creation).

Reply:
Fixed by separating the apply-order use case into a child class `TxSetInApplyOrder`, so not to mingle with the normal use case.